code/__HELPERS/AStar.dm

```diff
@@ -36,13 +39,22 @@ Actual Adjacent procs :
 	var/g		//A* movement cost variable
 	var/h		//A* heuristic variable
 	var/nt		//count the number of Nodes traversed
+	var/bf		//bitflag for dir to expand.Some sufficiently advanced motherfuckery
 
-/datum/PathNode/New(s,p,pg,ph,pnt)
+/datum/PathNode/New(s,p,pg,ph,pnt,_bf,_i)
 	source = s
 	prevNode = p
 	g = pg
 	h = ph
-	f = g + h
+	f = g + h*(1+SSpathfinder.tiew)
+	nt = pnt
+	bf = _bf
+
+/datum/PathNode/proc/setp(p,pg,ph,pnt)
+	prevNode = p
+	g = pg
+	h = ph
+	f = g + h*(1+SSpathfinder.tiew)
```

Comment:
if the tie breaker weight never changes just make it a define so you're not doing a global var lookup every time you want to use it.